algo/src/main/java/org/neo4j/graphalgo/impl/metaPathComputation/getSchema/GetSchema.java

```diff
@@ -14,41 +15,34 @@
 
     private HeavyGraph graph;
     private IntIntHashMap labelDictionary;//maybe change to array if it stays integer->integer
-    private IntIntHashMap reversedLabelDictionary;//also change to Array
+    private HashMap<Integer, Integer> reversedLabelDictionary;//also change to Array
     private int amountOfLabels;
     private int numberOfCores;
+    private PrintStream debugOut;
+    private long startTime;
+    private long endTime;
 
-    public GetSchema(HeavyGraph graph) {
+    public GetSchema(HeavyGraph graph) throws FileNotFoundException {
         this.graph = graph;
         this.amountOfLabels = graph.getAllLabels().size();
         this.labelDictionary = new IntIntHashMap();
-        this.reversedLabelDictionary = new IntIntHashMap();
+        this.reversedLabelDictionary = new HashMap<>();
         this.numberOfCores = Runtime.getRuntime().availableProcessors();
+
+        this.debugOut = new PrintStream(new FileOutputStream("Get_Schema_Debug.txt"));
     }
 
     public Result compute() {
-        ArrayList<HashSet<Pair>> schema = null;
-        boolean notYetComputed = false;
-
-        try {
-            FileInputStream fileIn = new FileInputStream("metagraph.ser");
-            ObjectInputStream in = new ObjectInputStream(fileIn);
-            schema = (ArrayList<HashSet<Pair>>) in.readObject();
-            in.close();
-            fileIn.close();
-        } catch (IOException i) {
-            notYetComputed = true;
-        } catch (ClassNotFoundException c) {
-            c.printStackTrace();
-            notYetComputed = true;
-        }
+        debugOut.println("START GET_SCHEMA");
 
-        if (notYetComputed) schema = computeSchema();
+        startTime = System.nanoTime();
+        ArrayList<HashSet<Pair>> schema = computeSchema();
+        endTime = System.nanoTime();
 
+        debugOut.println("FINISH GET_SCHEMA after " + (endTime - startTime) / 1000000 + " milliseconds");
         return new Result(schema, labelDictionary, reversedLabelDictionary);
     }
 
-
     private ArrayList<HashSet<Pair>> computeSchema() {
         initializeLabelDict();
```

Comment:
Maybe during the loading of the graph?

Reply:
In the Proc? Or somethere in neo4j's graphLoader?

Reply:
graphLoader